### ISSUE TITLE
Lint

### ISSUE THREAD

**Issue by ericawright** [mozilla/something-awesome#27]
- Add a linter so that we all follow the same coding conventions
https://github.com/eslint/eslint
https://github.com/stylelint/stylelint

- Protect master branch, and hook up Travis or Circle CI to check that it passes for PRs

**Comment by pdehaan** [comment]
For ESLint, probably something like this (at a minimum):

```yaml
env:
  browser: true
  es6: true
  node: true

extends:
  - eslint:recommended

rules:
  eqeqeq: error
  no-console: warn
  no-var: error
  one-var: [error, never]
  prefer-const: error
  quotes: [error, single]
```

**Comment by pdehaan** [comment]
I haven't used **stylelint** a bunch, but this seems to work as a minimal config (plus, latest version of **prettier** also formats CSS files):

```diff
  "devDependencies": {
    ...
    "prettier": "1.4.4",
+   "stylelint": "7.10.1",
+   "stylelint-config-standard": "16.0.0",
    ...
  },
  "scripts": {
+   "format:css": "prettier --write 'public/*.css'",
+   "lint:css": "stylelint 'public/*.css' --fix",
    ...
  }
```

And then a .stylelintrc config file of:

```yaml
extends:
  - stylelint-config-standard

rules:
  selector-list-comma-newline-after: null
```

**Comment by pdehaan** [comment]
Looking at this a bit closer, we _may_ want to add some basic HTML linting as well:

```sh
➜  something-awesome git:(master) $ npm run lint:html

> portal-alpha@1.0.0 lint:html /Users/pdehaan/dev/github/mozilla/something-awesome
> htmllint 'views/*.handlebars'

views/download.handlebars: line 19, col 7, tag is not closed
views/download.handlebars: line 23, col 9, the `alt` property must be set for image tags
views/download.handlebars: line 34, col 9, the `alt` property must be set for image tags
views/index.handlebars: line 18, col 30, the `alt` property must be set for image tags
views/index.handlebars: line 39, col 12, the `width` attribute is banned
views/index.handlebars: line 39, col 18, the `width` attribute is not double quoted
views/index.handlebars: line 40, col 12, the `width` attribute is banned
views/index.handlebars: line 40, col 18, the `width` attribute is not double quoted
views/index.handlebars: line 41, col 12, the `width` attribute is banned
views/index.handlebars: line 41, col 18, the `width` attribute is not double quoted
views/index.handlebars: line 42, col 12, the `width` attribute is banned
views/index.handlebars: line 42, col 18, the `width` attribute is not double quoted
views/index.handlebars: line 44, col 31, value must match the format: dash
views/index.handlebars: line 44, col 9, tag is not closed
views/index.handlebars: line 53, col 13, the id "upload-img" is already in use
views/index.handlebars: line 53, col 30, the `alt` property must be set for image tags
views/index.handlebars: line 65, col 9, the `alt` property must be set for image tags

[htmllint] found 17 errors out of 2 files
```

package.json:
```js
"lint:html": "htmllint 'views/*.handlebars'",
```

.htmllintrc:
```json
{
  "plugins": [],

  "attr-name-style": "dash",
  "indent-style": "spaces",
  "indent-width": 0,
  "id-class-style": "dash"
}
```

**Comment by ericawright** [comment]
I'm agreeable. though I'd skip the  `"indent-style": "spaces"` rule. and wouldn't `"indent-width": 0` allow no indents?

**Comment by pdehaan** [comment]
Setting `indent-width: 0` just disables that rule. Although I think it was an early bug in my config since the default is an insane 4 spaces. So setting `indent-width: 2` also works with no additional errors.

I can remove that `indent-style` rule too. I think the default is to allow spaces OR tabs, but not mixed in the same file, which is cool too.

Also, I can PR all these as soon as that ESLint PR lands so I can lazily avoid/reduce merge conflicts.

**Comment by pdehaan** [comment]
@ericawright: I think this is finally resolved via #77. I lack admin permissions on this repo to close other people's issues. But feel free to close if you think we've added enough (or too much) linting for meow.

**Comment by ericawright** [comment]
resolved
